### PR TITLE
[PR-18042] Add currency as required attribute for multi-currency sites

### PR DESCRIPTION
**Pull request by imnastasia**
Main PR [#8364](https://github.com/talkable/talkable/pull/8364)

## Demo

https://deploy-preview-270--talkable-docs.netlify.app/integration/custom/alternate_post_purchase.html?highlight=currency%20iso_code
https://deploy-preview-270--talkable-docs.netlify.app/web_hooks/event.html?highlight=currency%20iso_code
https://deploy-preview-270--talkable-docs.netlify.app/advanced_features/product_items.html?highlight=currency%20iso_code
https://deploy-preview-270--talkable-docs.netlify.app/advanced_features/events.html?highlight=currency%20iso_code
https://deploy-preview-270--talkable-docs.netlify.app/integration/custom/integration_components.html?highlight=currency%20iso_code

## Related Stories

[](https://talkable.atlassian.net/browse/PR-18042)

### CODE REVIEW

File: source/web_hooks/event.rst [hunk 1]
```diff
@@ -53,6 +54,7 @@ or using the :ref:`Talkable backend API <api_v2/origins>`.
      * **event_number** - unique identifier of the Event within the associated **event_category**
      * **customer_id** - unique external identifier of a customer who triggered the event
      * **subtotal** - optional monetary attribute of the Event
+     * **currency_iso_code** - currency ISO code of the Purchase
```
Comment (zhuravel) [inline source/web_hooks/event.rst:57]:
```suggestion
     * **currency_iso_code** - currency ISO code of the Event
```

File: source/samples/advanced_features/product_items.rst [hunk 2]
```diff
@@ -21,6 +21,7 @@
         order_number: '', // Required - Unique order number. Example: '100011'
         subtotal: '', // Required - Order subtotal (pre-tax, post-discount). Example: '23.97'
         coupon_code: '', // Coupon code that was used at checkout (pass multiple as an array). Example: 'SAVE20'
+        currency_iso_code: '', // It is required for multi-currency sites. Example: 'USD'
```
Comment (zhuravel) [inline source/samples/advanced_features/product_items.rst:24]:
```suggestion
        currency_iso_code: '', // Required for multi-currency sites. Example: 'USD'
```

File: source/integration/custom/integration_components.rst [hunk 3]
```diff
@@ -121,6 +122,7 @@ passed where division by zero or `null` value does not occur.
          order_number: '', // Required - Unique order number. Example: '100011'
          subtotal: '', // Required - Order subtotal (pre-tax, post-discount). Example: '23.97'
          coupon_code: '', // Coupon code that was used at checkout (pass multiple as an array). Example: 'SAVE20'
+         currency_iso_code: '', // It is required for multi-currency sites. Example: 'USD'
```
Comment (zhuravel) [inline source/integration/custom/integration_components.rst:125]:
```suggestion
         currency_iso_code: '', // Required for multi-currency sites. Example: 'USD'
```

File: source/integration/custom/alternate_post_purchase.rst [hunk 4]
```diff
@@ -24,6 +24,7 @@ Alternate Post Purchase Script for cart line item passing
         order_number: '', // Required - Unique order number. Example: '100011'
         subtotal: '', // Required - Order subtotal (pre-tax, post-discount). Example: '23.97'
         coupon_code: '', // Coupon code that was used at checkout (pass multiple as an array). Example: 'SAVE20'
+        currency_iso_code: '', // It is required for multi-currency sites. Example: 'USD'
```
Comment (zhuravel) [inline source/integration/custom/alternate_post_purchase.rst:27]:
```suggestion
        currency_iso_code: '', // Required for multi-currency sites. Example: 'USD'
```

**Review by zhuravel — CHANGES_REQUESTED** [review]

File: source/web_hooks/event.rst [hunk 5]
```diff
@@ -77,6 +79,7 @@ or using the :ref:`Talkable backend API <api_v2/origins>`.
        "customer_id": "865955119",
        "order_date": "2019-04-11T07:26:17.272-07:00",
        "coupon_code": "WHT28499",
+       "currency_iso_code": "USD",
```
[the remark this answers is not in the document]
Comment (imnastasia) [reply source/web_hooks/event.rst:82]: @zhuravel Correct me if I'm wrong, but in ParseParams we process any param inside the payload hash with 'o'/'origin' key

[the remark this answers is not in the document]
Comment (zhuravel) [reply source/web_hooks/event.rst:82]: @imnastasia Don’t see how `ParseParams` is related to Event Webhook.
[This serializer](https://github.com/talkable/talkable/blob/master/app/serializers/payload/event_serializer.rb) is used for the webhook and it does not contain `currency_iso_code`.

[the remark this answers is not in the document]
Comment (imnastasia) [reply source/web_hooks/event.rst:82]: @zhuravel oh I see. 
should I create a separate ticket to add it to the serializers and update docs, or just make a PR for adding it to the code without a separate ticket?

[the remark this answers is not in the document]
Comment (zhuravel) [reply source/web_hooks/event.rst:82]: @imnastasia 2

**Review by zhuravel — APPROVED** [review]

**Pull request merged**